source/Willow-Core/SingleSelectionWebView.class.st

```diff
@@ -0,0 +1,114 @@
+Class {
+	#name : #SingleSelectionWebView,
+	#superclass : #SingleSelectionWebViewBehavior,
+	#instVars : [
+		'commandToComponent',
+		'interactionInterpreter',
+		'identifierAssigner',
+		'optionCommand'
+	],
+	#category : #'Willow-Core-WebViews'
+}
+
+{ #category : #'instance creation' }
+SingleSelectionWebView class >> dropDownListApplying: aComponentCommand [
+
+	^ self prefixedBy: 'drop-down-list' applying: aComponentCommand
+]
+
+{ #category : #'instance creation' }
+SingleSelectionWebView class >> listBoxDisplayingAtOnce: aNumberOfElements [
+
+	^ self
+		listBoxDisplayingAtOnce: aNumberOfElements
+		applying: [ :listBox |  ]
+]
+
+{ #category : #'instance creation' }
+SingleSelectionWebView class >> listBoxDisplayingAtOnce: aNumberOfElements applying: aCommandComponent [
+
+	AssertionChecker
+		enforce: [ aNumberOfElements > 1 ]
+		because: 'A list box must display more than one element'.
+
+	^ self
+		prefixedBy: 'list-box'
+		applying:
+			(SetComponentSizeCommand setting: aNumberOfElements)
+				+ aCommandComponent
+]
+
+{ #category : #'private - instance creation' }
+SingleSelectionWebView class >> prefixedBy: aPrefix applying: aComponentCommand [
+
+	^ self new
+		initializePrefixedBy: aPrefix
+		applying: aComponentCommand
+]
+
+{ #category : #initialization }
+SingleSelectionWebView >> applyToEachOption: aCommand [
+
+	optionCommand := aCommand
+]
+
+{ #category : #private }
+SingleSelectionWebView >> configure: select unselectingAllOn: aCanvas [
+
+	^ availableElements ifNotEmpty: [ select apply: DeselectAllOptionsCommand new ]
+]
+
+{ #category : #accessing }
+SingleSelectionWebView >> identifierOn: aCanvas [
+
+	^ identifierAssigner identifierOn: aCanvas
+]
+
+{ #category : #initialization }
+SingleSelectionWebView >> initializePrefixedBy: aPrefix applying: aCommandToComponent [
```

Comment:
There's no need to keep the prefix. Both uses could say 'single-selection' and then this collaborator could be removed from the instance creation methods.